### PR TITLE
Oe/task/table body

### PR DESCRIPTION
**Pull request by JenniferHIM**

### CODE REVIEW

File: src/components/TableMatrix/TableBody/TableBody.module.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.banner {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
```
Comment (ZamkovaTetiana) [inline src/components/TableMatrix/TableBody/TableBody.module.scss]: use mixin

File: src/function/index.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import {FC} from 'react';
+import {v4 as uuidv4} from 'uuid';
+
+export interface ICell {
+  Amount: number;
+  ID: string;
+}
+
+export const randomGenerator = (): ICell => ({
+  Amount: Math.floor(Math.random() * (999 - 100 + 1)) + 100,
```
Comment (ZamkovaTetiana) [inline src/function/index.tsx]: Math.floor(Math.random() * 1000)

File: src/function/index.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  ID: uuidv4(),
+});
+
+export const createMatrix = (settings: {columns: number; rows: number}): Array<Array<ICell>> => {
+  // if (settings.columns && settings.rows) {
```
Comment (ZamkovaTetiana) [inline src/function/index.tsx]: comment

File: src/function/index.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    for (let j = 0; j < rows; j++) {
+      acc += matrixRows[j][i].Amount;
+    }
+    columnAvg = [...columnAvg, Math.round(acc / Number(rows))];
+  }
+  return columnAvg;
+};
+
+export const findNearestCells = (cells: number, sortedMatrix: Array<ICell>, item: ICell): Array<ICell> => {
+  const copySortedMatrix = [...[], ...sortedMatrix];
```
Comment (ZamkovaTetiana) [inline src/function/index.tsx]: if you need to create a copy of the array, there is no need to write [...[], ...sortedMatrix], just [...sortedMatrix]

File: src/redux/matrix/actions.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  const sortedMatrix = matrixSort(matrix);
+  return {
+    type: types.SET_SETTINGS,
+    payload: {settings, matrix, sortedMatrix},
+  };
+};
+
+// export const createMatrix = () => ({
+//   type: types.CREATE_MATRIX,
+// });
```
Comment (ZamkovaTetiana) [inline src/redux/matrix/actions.tsx]: //

File: src/redux/matrix/reducer.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      };
+
+    case types.CREATE_MATRIX:
+      return {...state, matrixRows: [action.payload]};
+
+    case types.INCREMENT_CELL:
+      const newMatrixRows = state.matrixRows.map((row) => {
+        return row.map((item) => {
+          if (item.ID === action.payload.ID) {
+            // item.Amount += 1;
```
Comment (ZamkovaTetiana) [inline src/redux/matrix/reducer.ts]: //

File: src/redux/matrix/reducer.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+    case types.DELETE_ROW:
+      const newMatrix = [...state.matrixRows].splice(action.payload, 1);
+      return {
+        ...state,
+        sortedMatrix: [...matrixSort(newMatrix)],
+        //TODO
+        // ...state.sortedMatrix.filter((el) =>
+        //   state.matrixRows[action.payload].every((element) => element.ID !== el.ID)
+        // ),
```
Comment (ZamkovaTetiana) [inline src/redux/matrix/reducer.ts]: TODO what? Need to describe what you need to do in comment

Comment (JenniferHIM) [reply src/redux/matrix/reducer.ts]: already nothing))))

**Review by ZamkovaTetiana — APPROVED** [review]

**Pull request merged**